Unable to pull latest image of kaniko-executor on ppc64le platform

While pulling the latest image of kaniko-executor on ppc64le, I am getting the following error : 

 `Error: choosing an image from manifest list docker://gcr.io/kaniko-project/executor:v1.7.0: no image found in the manifest list for architecture ppc64le, variant "", OS Linux 

ie. 

```
# docker pull gcr.io/kaniko-project/executor:v1.7.0
Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
Trying to pull gcr.io/kaniko-project/executor:v1.7.0...
Error: choosing an image from manifest list docker://gcr.io/kaniko-project/executor:v1.7.0: no image found in manifest list for architecture ppc64le, variant "", OS linux

```

@imjasonh, I made the following changes : 

```
[root@rdr-odf-upg2-syd05-bastion-0 kaniko]# git diff
diff --git a/.github/workflows/images.yaml b/.github/workflows/images.yaml
index 09e3cbc2..6c1a9b9f 100644
--- a/.github/workflows/images.yaml
+++ b/.github/workflows/images.yaml
@@ -33,14 +33,14 @@ jobs:
         include:
         - image: executor
           dockerfile: ./deploy/Dockerfile
-          platforms: linux/amd64,linux/arm64,linux/s390x
+          platforms: linux/amd64,linux/arm64,linux/s390x,linux/ppc64le
           image-name: gcr.io/kaniko-project/executor
           tag: ${{ github.sha }}
           release-tag: latest
 
         - image: executor-debug
           dockerfile: ./deploy/Dockerfile_debug
-          platforms: linux/amd64,linux/arm64,linux/s390x
+          platforms: linux/amd64,linux/arm64,linux/s390x,linux/ppc64le
           image-name: gcr.io/kaniko-project/executor
           tag: ${{ github.sha }}-debug
           release_tag: debug
@@ -54,7 +54,7 @@ jobs:
 
         - image: warmer
           dockerfile: ./deploy/Dockerfile_warmer
-          platforms: linux/amd64,linux/arm64,linux/s390x
+          platforms: linux/amd64,linux/arm64,linux/s390x,linux/ppc64le
           image-name: gcr.io/kaniko-project/warmer
           tag: ${{ github.sha }}
           release-tag: latest
```
However while running `make images` command, I am getting following error :  

```
[root@rdr-odf-upg2-syd05-bastion-0 kaniko]# make images
docker build --platform linux/ppc64le --build-arg=GOARCH=ppc64le -t gcr.io/kaniko-project/executor:latest -f deploy/Dockerfile .
Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
WARN[0000] missing "TARGETARCH" build argument. Try adding "--build-arg TARGETARCH=<VALUE>" to the command line 
[1/3] STEP 1/12: FROM golang:1.17
? Please select an image: 
✔ docker.io/library/golang:1.17
Trying to pull docker.io/library/golang:1.17...
Getting image source signatures
Copying blob 9ecf28457895 done  
Copying blob 3666fa80bd5b done  
Copying blob c0d4b2a71bc2 done  
Copying blob 8199fa873cad done  
Copying blob 7c968dbebedd done  
Copying blob 91c8a760808c done  
Copying blob 7651f3f70ff5 done  
Copying config f06e318d69 done  
Writing manifest to image destination
Storing signatures
[1/3] STEP 2/12: WORKDIR /src
--> 4d0298feae3
[1/3] STEP 3/12: ARG TARGETARCH
--> dd91ae30e4e
[1/3] STEP 4/12: ENV GOARCH=$TARGETARCH
--> 1872e3298b2
[1/3] STEP 5/12: ENV CGO_ENABLED=0
--> c0135c7c07a
[1/3] STEP 6/12: ENV GOBIN=/usr/local/bin
--> 11a2daa0123
[1/3] STEP 7/12: RUN go install github.com/GoogleCloudPlatform/docker-credential-gcr@4cdd60d0f2d8a69bc70933f4d7718f9c4e956ff8
go: downloading github.com/GoogleCloudPlatform/docker-credential-gcr v1.5.1-0.20210909233216-4cdd60d0f2d8
go: downloading github.com/google/subcommands v1.2.0
go: downloading github.com/docker/cli v20.10.8+incompatible
go: downloading github.com/docker/docker-credential-helpers v0.6.4
go: downloading github.com/toqueteos/webbrowser v1.2.0
go: downloading golang.org/x/oauth2 v0.0.0-20210819190943-2bc19b11175f
go: downloading cloud.google.com/go v0.65.0
go: downloading golang.org/x/net v0.0.0-20210405180319-a5a99cb37ef4
go: downloading github.com/docker/docker v20.10.8+incompatible
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/sirupsen/logrus v1.8.1
go: downloading golang.org/x/sys v0.0.0-20210510120138-977fb7262007
--> a0c8dc24b57
[1/3] STEP 8/12: RUN go install github.com/awslabs/amazon-ecr-credential-helper/ecr-login/cli/docker-credential-ecr-login@69c85dc22db6511932bbf119e1a0cc5c90c69a7f # v0.6.0
go: downloading github.com/awslabs/amazon-ecr-credential-helper/ecr-login v0.0.0-20211215200129-69c85dc22db6
go: downloading github.com/awslabs/amazon-ecr-credential-helper v0.6.0
go: downloading github.com/docker/docker-credential-helpers v0.6.3
go: downloading github.com/aws/aws-sdk-go-v2 v1.7.1
go: downloading github.com/aws/aws-sdk-go-v2/config v1.5.0
go: downloading github.com/aws/aws-sdk-go-v2/service/ecr v1.4.1
go: downloading github.com/aws/aws-sdk-go-v2/service/ecrpublic v1.4.1
go: downloading github.com/aws/smithy-go v1.6.0
go: downloading github.com/sirupsen/logrus v1.4.2
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/aws/aws-sdk-go-v2/credentials v1.3.1
go: downloading github.com/aws/aws-sdk-go-v2/feature/ec2/imds v1.3.0
go: downloading github.com/aws/aws-sdk-go-v2/internal/ini v1.1.1
go: downloading github.com/aws/aws-sdk-go-v2/service/sso v1.3.1
go: downloading github.com/aws/aws-sdk-go-v2/service/sts v1.6.0
go: downloading github.com/jmespath/go-jmespath v0.4.0
go: downloading golang.org/x/sys v0.0.0-20210423082822-04245dca01da
go: downloading github.com/aws/aws-sdk-go-v2/service/internal/presigned-url v1.2.1
--> 47c42c07ea6
[1/3] STEP 9/12: RUN go install github.com/chrismellard/docker-credential-acr-env@09e2b5a8ac86c3ec347b2473e42b34367d8fa419
go: downloading github.com/chrismellard/docker-credential-acr-env v0.0.0-20210203204924-09e2b5a8ac86
go: downloading github.com/spf13/viper v1.7.1
go: downloading github.com/spf13/cobra v1.0.0
go: downloading github.com/Azure/go-autorest/autorest/azure/auth v0.5.2
go: downloading github.com/Azure/azure-sdk-for-go v46.4.0+incompatible
go: downloading github.com/Azure/go-autorest/autorest v0.11.8
go: downloading github.com/Azure/go-autorest/autorest/adal v0.9.4
go: downloading github.com/Azure/go-autorest v14.2.0+incompatible
go: downloading github.com/fsnotify/fsnotify v1.4.9
go: downloading github.com/hashicorp/hcl v1.0.0
go: downloading github.com/magiconair/properties v1.8.4
go: downloading github.com/mitchellh/mapstructure v1.3.3
go: downloading github.com/pelletier/go-toml v1.8.1
go: downloading github.com/spf13/afero v1.4.1
go: downloading github.com/spf13/cast v1.3.1
go: downloading github.com/spf13/jwalterweatherman v1.1.0
go: downloading github.com/spf13/pflag v1.0.5
go: downloading github.com/subosito/gotenv v1.2.0
go: downloading gopkg.in/ini.v1 v1.62.0
go: downloading gopkg.in/yaml.v2 v2.3.0
go: downloading github.com/Azure/go-autorest/autorest/azure/cli v0.4.1
go: downloading github.com/dimchansky/utfbom v1.1.0
go: downloading github.com/Azure/go-autorest/logger v0.2.0
go: downloading github.com/Azure/go-autorest/tracing v0.6.0
go: downloading github.com/Azure/go-autorest/autorest/date v0.3.0
go: downloading github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: downloading golang.org/x/crypto v0.0.0-20201002170205-7f63de1d35b0
go: downloading golang.org/x/sys v0.0.0-20201005172224-997123666555
go: downloading golang.org/x/text v0.3.3
--> 5d68566548e
[1/3] STEP 10/12: RUN mkdir -p /kaniko/.docker
--> cba0914d42f
[1/3] STEP 11/12: COPY . .
--> 176af4ac956
[1/3] STEP 12/12: RUN     --mount=type=cache,target=/root/.cache/go-build     --mount=type=cache,target=/go/pkg     make GOARCH=$TARGETARCH
[2/3] STEP 1/2: FROM debian:buster-slim AS certs
Resolved "debian" as an alias (/etc/containers/registries.conf.d/000-shortnames.conf)
Trying to pull docker.io/library/debian:buster-slim...
Error: error building at STEP "RUN --mount=type=cache,target=/root/.cache/go-build --mount=type=cache,target=/go/pkg make GOARCH=$TARGETARCH": error resolving mountpoints for container "d0cd790f174de02a97b654c87fd068d7a924bdc407f23834a61448081b05f0c8": invalid filesystem type "cache"
make: *** [Makefile:97: images] Error 125
```
Not sure, why ?

I manually added `DOCKER_BUILDKIT=1` in Makefile . But it is still failing with the same error. 

```
[root@rdr-odf-upg2-syd05-bastion-0 kaniko]# git diff Makefile
diff --git a/Makefile b/Makefile
index 1a3b87c5..3e5a17e8 100644
--- a/Makefile
+++ b/Makefile
@@ -93,6 +93,7 @@ k8s-executor-build-push:
        docker push $(REGISTRY)/executor:latest
 
 .PHONY: images
+images: DOCKER_BUILDKIT=1
 images:
        docker build ${BUILD_ARG} --build-arg=GOARCH=$(GOARCH) -t $(REGISTRY)/executor:latest -f deploy/Dockerfile .
        docker build ${BUILD_ARG} --build-arg=GOARCH=$(GOARCH) -t $(REGISTRY)/executor:debug -f deploy/Dockerfile_debug .
```

error : 

```
Error: error building at STEP "RUN --mount=type=cache,target=/root/.cache/go-build --mount=type=cache,target=/go/pkg make GOARCH=$TARGETARCH": error resolving mountpoints for container "1a3cdd0a72265cedec09ed66c09a0ee736e6f7fdafac637308114ec53d8d4b2e": invalid filesystem type "cache"
make: *** [Makefile:98: images] Error 125
```

Jason, with docker version : 

```
[root@rdr-odf-upg2-syd05-bastion-0 kaniko]# docker version
Client: Docker Engine - Community
 Version:           v20.10.12
 API version:       1.41
 Go version:        go1.16.12
 Git commit:        e91ed57
 Built:             Tue Dec 14 10:12:12 2021
 OS/Arch:           linux/ppc64le
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          v20.10.12
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.12
  Git commit:       459d0df
  Built:            Tue Dec 14 10:10:07 2021
  OS/Arch:          linux/ppc64le
  Experimental:     false
 containerd:
  Version:          1.4.12
  GitCommit:        7b11cfaabd73bb80907dd23182b9347b4245eb5d
 runc:
  Version:          1.0.2
  GitCommit:        v1.0.2-0-g52b36a2
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

```

and go version -> `go version go1.17.6 linux/ppc64le'

image build proceeded, but it is still failing with new error : 

```
[root@rdr-odf-upg2-syd05-bastion-0 kaniko]# make images
DOCKER_BUILDKIT=1 docker build  --build-arg=GOARCH=ppc64le -t gcr.io/kaniko-project/executor:latest -f deploy/Dockerfile .
[+] Building 161.8s (10/23)                                                                                                                                                                                 
 => [internal] load build definition from Dockerfile                                                                                                                                                   0.0s
 => => transferring dockerfile: 92B                                                                                                                                                                    0.0s
 => [internal] load .dockerignore                                                                                                                                                                      0.0s
 => => transferring context: 2B                                                                                                                                                                        0.0s
 => [internal] load metadata for docker.io/library/debian:buster-slim                                                                                                                                  1.8s
 => [internal] load metadata for docker.io/library/golang:1.17                                                                                                                                         1.8s
 => [stage-0 1/8] FROM docker.io/library/golang:1.17@sha256:301609ebecc0ec4cd3174294220a4d9c92aab9015b3a2958297d7663aac627a1                                                                           0.0s
 => [internal] load build context                                                                                                                                                                      0.3s
 => => transferring context: 784.92kB                                                                                                                                                                  0.3s
 => [certs 1/2] FROM docker.io/library/debian:buster-slim@sha256:f6e5cbc7eaaa232ae1db675d83eabfffdabeb9054515c15c2fb510da6bc618a7                                                                      0.0s
 => CACHED [stage-0 2/8] WORKDIR /src                                                                                                                                                                  0.0s
 => ERROR [stage-0 3/8] RUN go install github.com/GoogleCloudPlatform/docker-credential-gcr@4cdd60d0f2d8a69bc70933f4d7718f9c4e956ff8                                                                 159.8s
 => CACHED [certs 2/2] RUN   apt update &&   apt install -y ca-certificates &&   cat /etc/ssl/certs/* > /ca-certificates.crt                                                                           0.0s
------                                                                                                                                                                                                      
 > [stage-0 3/8] RUN go install github.com/GoogleCloudPlatform/docker-credential-gcr@4cdd60d0f2d8a69bc70933f4d7718f9c4e956ff8:                                                                              
#7 6.526 go: downloading github.com/GoogleCloudPlatform/docker-credential-gcr v1.5.1-0.20210909233216-4cdd60d0f2d8                                                                                          
#7 159.5 go install: github.com/GoogleCloudPlatform/docker-credential-gcr@4cdd60d0f2d8a69bc70933f4d7718f9c4e956ff8: github.com/GoogleCloudPlatform/docker-credential-gcr@v1.5.1-0.20210909233216-4cdd60d0f2d8: read tcp 172.17.0.2:33226->142.251.46.177:443: read: connection reset by peer
------
executor failed running [/bin/sh -c go install github.com/GoogleCloudPlatform/docker-credential-gcr@4cdd60d0f2d8a69bc70933f4d7718f9c4e956ff8]: exit code: 1
make: *** [Makefile:97: images] Error 1
```

Not sure what I am missing now ?

--- a/.github/workflows/images.yaml
+++ b/.github/workflows/images.yaml
@@ -33,7 +33,7 @@ jobs:
         include:
         - image: executor
           dockerfile: ./deploy/Dockerfile
-          platforms: linux/amd64,linux/arm64,linux/s390x
+          platforms: linux/amd64,linux/arm64,linux/s390x,linux/ppc64le
           image-name: gcr.io/kaniko-project/executor
           tag: ${{ github.sha }}
           release-tag: latest
@@ -54,7 +54,7 @@ jobs:
 
         - image: warmer
           dockerfile: ./deploy/Dockerfile_warmer
-          platforms: linux/amd64,linux/arm64,linux/s390x
+          platforms: linux/amd64,linux/arm64,linux/s390x,linux/ppc64le
           image-name: gcr.io/kaniko-project/warmer
           tag: ${{ github.sha }}
           release-tag: latest

